### ISSUE TITLE
Shipping methods with multiple selections not displaying

### ISSUE THREAD

**Issue by DasWookie** [gburton/Responsive-osCommerce#526]
I had installed the UPS Shipping module, and wasn't getting the elections showing up. I then installed a clean install with no modifications of 2.3.4 Gold and then configured the USPS module, and also am not getting any selections showing up. It seems that the any shipping methods with multiple selections for choices are not displaying. Shipping methods with one option to pick are all working fine (flat, zones, etc) however.

**Comment by DasWookie** [comment]
Note: The reason why I attempted to use the USPS module for debugging was that the USPS module is built in and returns multiple elections. I have problems with the USPS module as well on my modded install, so that's when I saw the connection between the two. Both return more than one value for a customer to pick from. That's when it dawned on me that my problem might not be with the UPS module, but with any shipping modules which return more than one choice. Because the USPS module is a native module, and I could test with it on a baseline install, if I debugged with the USPS module and got it working, most likely the UPS module would then also be fixed.

**Comment by BrockleyJohn** [comment]
According to @kymation the USPS module does not work and has not done so for several years.

**Comment by DasWookie** [comment]
I don't actually use USPS shipping. I was just trying to use it as a means of trying to debug why the UPS module wasn't displaying shipping options, even tho I was successfully retrieving a quote for multiple options. I posted on the forum support thread, but generally whenever I have a problem I'm trying to debug, I get told it's specific to 2.3.4 BS GOLD compatibility and not an issue with the add-on itself... thus my thinking I could try the USPS module that was built in on BS GOLD.

My main issue is I can't get any shipping quotes to display in BS GOLD even tho the XML I get back from UPS is good and has valid data.

My request on the forum has been silent for 2 weeks now, so that's why I started trying to debug if it is related to BS GOLD http://forums.oscommerce.com/topic/49382-new-ups-xml-shipping-module-available/?p=1752800 as I'm otherwise at a complete loss as to what the problem might be and I can't find anyone else (at least on the forum) who's using the UPS module and BS GOLD who have any idea as to what the problem might be. The UPS add-on http://addons.oscommerce.com/info/9317 is installed correctly (I've checked it six ways from Sunday) and like I said, I'm getting a valid XML return from UPS... just never get any election that makes it to the user to pick. Shipping methods that only have one election, work fine.

![post-341041-0-84060100-1485721984](https://cloud.githubusercontent.com/assets/9283270/22857418/3f23ca08-f06a-11e6-9f71-3b95fab56899.jpg)

I'm grasping at straws now to try and get this working. It's the final remaining issue to taking my store live! :(

**Comment by BrockleyJohn** [comment]
I can see why you want UPS with that colour scheme ;)

Are you using the UPS addon unchanged?

**Comment by BrockleyJohn** [comment]
If you edit the module quote method and put
`echo '<pre>';
print_r($this->quotes);
echo '</pre>';`
just before 
`        return $this->quotes;`
on line 396, what do you get when you try again?

**Comment by gburton** [comment]
I'm not sure what you are asking. Are you stating that shipping modules that present multiple options for selection - that these multiple options do not show?

**Comment by DasWookie** [comment]
I've got it sorted out, finally! The problem was multifold, but the end result was I got overzealous on nuking pass by reference in the xml.php for the UPS shipping module. I'm getting quotes back successfully into the UI, finally!

@BrockleyJohn  you set me on the right path. The return quotes was empty, so I had to start digging in deeper and deeper until I finally found the problem in the xml.php, then slowly claw my way back to the surface one function call at a time. It was finally when I was trying to debug the xml-parse call when I noticed on the function spec page here:
http://php.net/manual/en/function.xml-parse.php

a curious snippet:
`parser
A reference to the XML parser to use.`

where it clicked. "Reference to", but any time the php hits a reference to the throws fatal? Doesn't it? Uhmmmm. So, I fell back to the initial xml.php and started removing only the & references that threw fatal one page submit at a time, until suddenly... "WHOAH!!! Quotes made it to the UI!!!"

So, issue is resolved. Not a BS Gold problem. The red herring of USPS and UPS both having multiple options sent me down the wrong path initially...

I'll close the issue out.
